src/include/miopen/find_solution.hpp

```diff
@@ -121,8 +121,6 @@ template <class Solver, class Context, class Db>
 ConvSolution
 FindSolution(Solver s, const Context& context, Db& db, const AnyInvokeParams& invoke_ctx)
 {
-    static_assert(std::is_empty<Solver>{} && std::is_trivially_constructible<Solver>{},
-                  "Solver must be stateless");
```

Comment:
You should be able to at least check that `sizeof(Solver) == sizeof(SolverBase)`

Reply:
Code was changed as you proposed, thanks